common-practices-tools/contribution/contrib-first.md

```diff
@@ -0,0 +1,29 @@
+# Contrib First
+
+Whenever we are building something that could be of use by more than one project or client, we build it as contributed work first, if our contract and security concerns allow for it. Contrib First is an approach we take with both patches to FOSS software and new contributions.
+
+## Rationale for contrib first
+
+-   **Fiscal responsibility** - Building it and contributing it means that other government agencies will never have to pay to build the same thing twice. This helps agencies comply with Federal Source Code Policy: Achieving Efficiency, Transparency, and Innovation through Reusable and Open Source Software [OMB Memorandum M-16-21](https://obamawhitehouse.archives.gov/sites/default/files/omb/memoranda/2016/m_16_21.pdf)
+-   **Reusability** - CivcActions other clients can benefit from work that was already done.
+-   **Security** - contributing our work to an open source project like Drupal means it may receive security coverage by the Drupal security team and the public.
```

Comment:
Security coverage (at least in terms of Drupal) doesn't actually lead to detection of any new security issues - it's focus is really only on coordinating fixes (completed by project maintainers) for issues that are reported by external users and then notifying other users.
This is a benefit of course, but I am not sure it necessarily adds security on it's own. I would consider starting out with the "[many eyes](https://en.wikipedia.org/wiki/Linus%27s_law)" argument. Essentially, by having many users it's more likely that security vulnerabilities will be detected over time (if it remains private, this is much less likely). Combining this with vulnerability reporting/notification infrastructure (like Drupal, but NVD is also a great example) you get the benefit of a more secure site with minimal additional effort.

Reply:
I adjusted this to include that.